runtime/lua/vim/lsp/util.lua

```diff
@@ -1354,71 +1354,55 @@ function M.stylize_markdown(bufnr, contents, opts)
   return stripped
 end
 
----@private
---- Creates autocommands to close a preview window when events happen.
----
----@param events table list of events
----@param winnr number window id of preview window
----@param bufnrs table list of buffers where the preview window will remain visible
----@see |autocmd-events|
-local function close_preview_autocmd(events, winnr, bufnrs)
-  local augroup = 'preview_window_' .. winnr
-
-  -- close the preview window when entered a buffer that is not
-  -- the floating window buffer or the buffer that spawned it
-  vim.cmd(string.format(
-    [[
-    augroup %s
-      autocmd!
-      autocmd BufEnter * lua vim.lsp.util._close_preview_window(%d, {%s})
-    augroup end
-  ]],
-    augroup,
-    winnr,
-    table.concat(bufnrs, ',')
-  ))
-
-  if #events > 0 then
-    vim.cmd(string.format(
-      [[
-      augroup %s
-        autocmd %s <buffer> lua vim.lsp.util._close_preview_window(%d)
-      augroup end
-    ]],
-      augroup,
-      table.concat(events, ','),
-      winnr
-    ))
-  end
-end
-
 ---@private
 --- Closes the preview window
 ---
 ---@param winnr number window id of preview window
 ---@param bufnrs table|nil optional list of ignored buffers
-function M._close_preview_window(winnr, bufnrs)
+local function close_preview_window(winnr, bufnrs)
   vim.schedule(function()
     -- exit if we are in one of ignored buffers
     if bufnrs and vim.tbl_contains(bufnrs, api.nvim_get_current_buf()) then
       return
     end
 
     local augroup = 'preview_window_' .. winnr
-    vim.cmd(string.format(
```

Comment:
@glepnir I don't think this block you removed is equivalent to the line you replaced it with. Locally having just updated I'm seeing error
```
Error executing vim.schedule lua callback: ...vim/HEAD-782f726/share/nvim/runtime/lua/vim/lsp/util.lua:1370: Vim:E367: No such group: "preview_window_1024"                                     
stack traceback:                                                                                                                                                                                
        [C]: in function 'nvim_del_augroup_by_name'                                                                                                                                             
        ...vim/HEAD-782f726/share/nvim/runtime/lua/vim/lsp/util.lua:1370: in function <...vim/HEAD-782f726/share/nvim/runtime/lua/vim/lsp/util.lua:1363> 
```
Sorry for commenting on a closed issue, but since it's only a few hours old, I thought it would be easiest to see/remember this way.

I think (but am not 100% sure) that this block first *created* the augroup and then cleared any of it's contained commands, then the next line deletes the group.

It's easiest to see this issue if you create a `CursorHold` autocommand that triggers `vim.diagnostic.open_float()` you'll see this error repeated on every cursor hold

Reply:
Because of how `:augroup` command works, you first have to remove the autocommands inside (`aug foo | au! | aug end`), and only then you can remove the group (`aug! foo`).

`nvim_del_augroup_by_name` removes autocommands automatically, but it can throw this error when the group was not found. Errors should be ignored here: `pcall(api.nvim_del_augroup_by_name, augroup)`